core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/identity/DefaultJWTGenerator.java

```diff
@@ -36,22 +40,30 @@ final class DefaultJWTGenerator implements JWTGenerator {
 	private final String issuer;
 	private final String emailClaimProperty;
 	private final String permissionsClaimProperty;
+	private final InstantSource instantSource;
 
-	public DefaultJWTGenerator(final Algorithm algorithm, final String issuer, final String emailClaimProperty, final String permissionsClaimProperty) {
+	public DefaultJWTGenerator(final Algorithm algorithm, final String issuer, final String emailClaimProperty, final String permissionsClaimProperty, InstantSource instantSource) {
 		this.algorithm = checkNotNull(algorithm);
 		this.issuer = issuer;
 		this.emailClaimProperty = emailClaimProperty;
 		this.permissionsClaimProperty = permissionsClaimProperty;
+		this.instantSource = checkNotNull(instantSource);
 	}
 	
 	@Override
-	public String generate(String email, Map<String, Object> claims) {
+	public String generate(String email, Map<String, Object> claims, String expiration) {
+		
 		Builder builder = JWT.create()
 				.withIssuer(issuer)
 				.withSubject(email)
 				.withIssuedAt(new Date())
 				.withClaim(emailClaimProperty, email);
 		
+		if (!Strings.isNullOrEmpty(expiration)) {
+			TimeValue expirationTimeValue = TimeValue.parseTimeValue(expiration, "expiration");
```

Comment:
A check for non-negative and "practical" expiration intervals could be added here in the future 🔮

Reply:
Added a test case and the check to the generator.